[#11] feat: 가게, 메뉴 기능 구현

yjll1019 opened the pull request.

yjll1019 committed:
chore: DB 관련 설정 추가

yjll1019 committed:
feat: 가게 등록 기능 구현

yjll1019 committed:
chore: test DB 설정 및 데이터 추가

yjll1019 committed:
feat: 가게 조회 기능 구현

yjll1019 committed:
chore: open-in-view false 설정 해제, data.sql 데이터 추가

yjll1019 committed:
chore: webflux, mysql-connector 의존성 추가

yjll1019 committed:
feat: 가게의 메뉴 추가 기능 구현

yjll1019 committed:
refactor: controller 패키지 수정

yjll1019 committed:
refactor: Money 객체 생성, 도메인 객체의 setter 삭제

yjll1019 committed:
refactor: equals(), hashCode(), toString() 재정의

yjll1019 committed:
refactor: MenuAssembler, ShopAssembler 생성

parker1609 commented on build.gradle:
```diff
@@ -22,8 +22,10 @@ repositories {
 dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
     implementation 'org.springframework.boot:spring-boot-starter-web'
+    implementation 'org.springframework.boot:spring-boot-starter-webflux'
```
`testImplementation`으로 변경해주세요. 하지만 OAuth쓰면 필요함

parker1609 commented on src/main/java/com/eattogether/heytogether/domain/Menu.java:
```diff
@@ -0,0 +1,78 @@
+// 특정 가게의 특정 메뉴
+@Entity
+public class Menu {
+    @Id
```
클래스 시작할 때 한 줄 띄우기

parker1609 commented on src/main/java/com/eattogether/heytogether/domain/Menu.java:
```diff
@@ -0,0 +1,78 @@
+@Entity
+public class Menu {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
`GenerationType.IDENTITY`를 `.AUTO`로 변경하면 어떨까~

parker1609 commented on src/main/java/com/eattogether/heytogether/domain/Place.java:
```diff
@@ -0,0 +1,63 @@
+import javax.persistence.*;
+
+@Entity
+public class Place {
```
엔티티보다는 VO로 만드는 것이 좋을 것 같아요.

parker1609 commented on src/main/java/com/eattogether/heytogether/domain/Menu.java:
```diff
@@ -0,0 +1,78 @@
+
+    @ManyToOne(cascade = CascadeType.ALL)
+    @JoinColumn(name = "shop_id")
+    @JsonBackReference
```
양방향으로 설정된 엔티티가 Service에서 Controller로 엔티티 자체를 던지기 때문에 무한 재귀가 발생합니다.... 이를 해결하려고 이 어노테이션을 썼는데, Service와 Controller가 통신할 때는 모두 DTO로 만들어서 던지는 것이 어떨까요?

parker1609 commented on src/main/java/com/eattogether/heytogether/domain/Shop.java:
```diff
@@ -0,0 +1,87 @@
+    @AttributeOverrides(@AttributeOverride(name = "value", column = @Column(name = "minimumOrderPrice")))
+    private Money minimumOrderPrice;
+
+    @OneToMany(mappedBy = "shop", cascade = CascadeType.MERGE)
```
`CascadeType.All`로 변경하기

parker1609 commented on src/test/java/com/eattogether/heytogether/web/controller/ShopApiControllerTest.java:
```diff
@@ -0,0 +1,57 @@
+    private WebTestClient webTestClient;
+
+    @Test
+    @DisplayName("가게 생성")
```
최대한 설명을 자세하게 남긴다.(Given When Then 에 맞추면 좋음)

parker1609 commented on src/test/java/com/eattogether/heytogether/web/controller/ShopApiControllerTest.java:
```diff
@@ -0,0 +1,57 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@AutoConfigureWebTestClient
+class ShopApiControllerTest {
```
`ShopApiAcceptanceTest` 이름으로 변경한다.

parker1609 reviewed (commented):
5명 다같이 리뷰 완료하였습니다. 반영부탁드립니다!

yjll1019 committed:
[#11] refactor: 개행 컨벤션

yjll1019 committed:
[#11] refactor: 불필요한 테스트 클래스 삭제

yjll1019 committed:
[#11] refactor: 피드백 반영

- gradle impl -> testImpl
- GenerationType.Identity -> AUTO
- CascadeType.Merge -> ALL
- ShopApiControllerTest -> ShopApiAcceptanceTest
- @DisplayName 설명 추가

yjll1019 committed:
[#11] refactor: Place를 Entity에서 VO로 변경

yjll1019 committed:
[#11] chore: hibernate id 생성 옵션 설정 추가

yjll1019 committed:
[#11] refactor: MenuInfoDto, ShopInfoDto 생성

yjll1019 committed:
[#11] chore: junit-jupiter-api 의존성 추가

yjll1019 committed:
[#11] test: MenuServiceTest, ShopServiceTest 추가

yjll1019 committed:
[#11] refactor: @JsonBackReference, @JsonManagedReference 어노테이션 삭제

parker1609 commented:
수고하셨습니다! 👍
